src/rx.rs

```diff
@@ -170,6 +177,13 @@ impl<'a> DerefMut for RxPacket<'a> {
 
 impl<'a> Drop for RxPacket<'a> {
     fn drop(&mut self) {
+        // "Preceding reads and writes cannot be moved past subsequent writes."
+        #[cfg(feature = "stm32f7xx-hal")]
+        atomic::fence(Ordering::Release);
+
+        #[cfg(not(feature = "stm32f7xx-hal"))]
+        atomic::compiler_fence(Ordering::Release);
+
         self.entry.desc_mut().set_owned();
```

Comment:
I assume the fence before setting the owned flag on RX descriptors is to ensure our reads of the descriptor and buffer are finished before we allow the DMA engine to start writing to it, but do we also want a fence after setting the owned flag on F7 to ensure the write is flushed through the store buffer? Perhaps inside the `set_owned()` method?

Reply:
Hmm, I placed a fence inside demand poll because of that, but it makes sense to have it on drop too, to make the buffer available for the DMA as fast as possible, but it would need to be a `SeqCst`. I'm not sure about placing it at `set_owned` itself, since it's not always needed there, especially because we call it a lot during the descriptors' setup.

All the other times that we call `set_owned` (when needing the fence) we already call `demand_poll` which has a fence. So, I'm compelled to put the fence (`SeqCst`) on the `drop` itself, what do you think ?

Reply:
When isn't it needed on `set_owned`? I haven't totally understood the store buffer semantics but it seems like we always want the OWNED bit to become visible to the DMA right after we set it, even if we're just setting it because there was an error in a received descriptor or something so we're throwing it away. Otherwise we don't know how long the write will stay in the store buffer before being made visible to the DMA, which might be an issue if we only have a couple of descriptors allocated.

Reply:
During the [initial setup](https://github.com/stm32-rs/stm32-eth/blob/669d03e73a79ceb6ce3ca82ea0d983944b0d8906/src/rx.rs#L214-L227), since the DMA isn't active.

Reply:
Although, I'm starting to think that having all the necessary fences in `set_owned` is worth it, because it seems to make it less error prone, a few DMB instructions during setup doesn't look bad, what do you think ?

Reply:
Yep, I see what you mean. Given as setup happens once I'm not really worried about a handful of DMB calls. If it was happening on every packet reception it might be a different balance of concerns. Given the simplicity and correctness of just putting it in `set_owned` that does seem appealing. What about for the Tx side?

Reply:
I will do the same on Tx, the fences will pretty much be moved from `send` and `demand_poll` to only `set_owned`.